src/main/scala/nl/wbaa/gargoyle/proxy/providers/AuthorizationProvider.scala

```diff
@@ -0,0 +1,8 @@
+package nl.wbaa.gargoyle.proxy.providers
+
+/**
+ * Interface for security provider implementations.
+ */
+trait AuthorizationProvider {
+  def isAuthorized(accessMode: String, path: String, username: String): Boolean = true
```

Comment:
I dont understand this. You are missing a lot of info that is part of the authorization context. AccessMode, Path, and Username are not enough. I also dont understand the extra abstraction?

Reply:
Agreed that we shouldn't over generalise, goal of these classes is to just provide a skeleton so we at least have some separation on the parts we're working on and not run into too many merge conflicts.
Indeed I would just use these classes to connect to the relevant services (e.g. Ranger) immediately and not use them as abstact bases.

Reply:
Also this comment was meant for the complete review, wrong box ;)